lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -136,10 +136,20 @@ namespace {
             if (requiresUnsharedVertices) {
                 const VtIntArray& faceVertexIndices = topology.GetFaceVertexIndices();
                 if (numVertices == faceVertexIndices.size()) {
+                    unsigned int dataSize = primvarData.size();
                     for (size_t v = 0; v < numVertices; v++) {
                         SRC_TYPE* pointer = reinterpret_cast<SRC_TYPE*>(
                             reinterpret_cast<float*>(&vertexBuffer[v]) + channelOffset);
-                        *pointer = primvarData[faceVertexIndices[v]];
+                        unsigned int index = faceVertexIndices[v];
+                        if (index < dataSize) {
+                            *pointer = primvarData[faceVertexIndices[v]];
+                        } else {
+                            TF_DEBUG(HDVP2_DEBUG_MESH).Msg("Invalid face vertex index '%s': "
+                                                    "primvar %s has %zu elements, while its topology "
+                                                    "references only upto element index %zu.\n",
+                                                    rprimId.asChar(), primvarName.GetText(),
+                                                    primvarData.size(), numVertices);                            
+                        }
```

Comment:
Stealing this error message from the block of code just a bit down from here in the file where we were looking at almost the exact same thing.